### PR TITLE
MAGECLOUD-5490: Capture MySQL statistics

### PR DESCRIPTION
**Pull request by shiftedreality**
### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-5490

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: Model/Cron/DbHealthList.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use Magento\Framework\App\DeploymentConfig;
+use Magento\Framework\App\ResourceConnection;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Logs DB health status.
+ */
+class DbHealthList
+{
+    const CONFIG_KEY = 'dev/debug/cloud_logging';
```
Comment (NadiyaS) [inline Model/Cron/DbHealthList.php]: how customer can manage this configuration?

Comment (shiftedreality) [reply Model/Cron/DbHealthList.php]: Good question 😄 

Added

File: Model/Cron/DbHealth.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        if (!$this->deploymentConfig->get(ConstantList::CONFIG_PATH_LOG_DB_HEALTH)) {
+            return;
+        }
+
+        $this->logger->debug('Process list', [
+            'data' => $this->connection->query('SHOW PROCESSLIST')->fetchAll(),
+        ]);
+        $this->logger->debug('InnoDB Status', [
+            'data' => $this->connection->query('SHOW ENGINE INNODB STATUS')->fetchAll()
```
Comment (kandy) [inline Model/Cron/DbHealth.php:63]: Please be aware that "SHOW ENGINE INNODB STATUS" required additional rights (SUPER) for execute. It's not available on PRO for default credentials

Comment (shiftedreality) [reply Model/Cron/DbHealth.php:63]: Yes, you're correct:

```
ERROR 1227 (42000): Access denied; you need (at least one of) the PROCESS privilege(s) for this operation
```

Do you think of any other way to track this data?

File: Console/Command/ConfigSetLogDbHealthCommand.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Changes DB health logging status
+ */
+class ConfigSetLogDbHealthCommand extends Command
+{
+    const NAME = 'config:set:log-db-health';
```
Comment (NadiyaS) [inline Console/Command/ConfigSetLogDbHealthCommand.php:24]: Why don't you add this to System configurations in Admin panel?
It also will be managed by `config:set` CLI command

Comment (shiftedreality) [reply Console/Command/ConfigSetLogDbHealthCommand.php:24]: The usage of this feature will be very limited, I expect it only to be enabled for Customers with deadlocks for debugging purposes

Comment (NadiyaS) [reply Console/Command/ConfigSetLogDbHealthCommand.php:24]: Also, customers who have RO FS, cannot use this function as it writes only to file.

Comment (shiftedreality) [reply Console/Command/ConfigSetLogDbHealthCommand.php:24]: It writes to `var/log` which is writable

Comment (NadiyaS) [reply Console/Command/ConfigSetLogDbHealthCommand.php:24]: `var/log` ? I am talking about the value of this config, not the logs which you write

**Pull request closed without merging**